### PR TITLE
Add default IREE ABI pass.

### PR DESCRIPTION
**Pull request by jpienaar**
Add reflection info by default if not set along *HLO and TOSA import paths. This retains the previous import path behavior that is getting deleted.

### CODE REVIEW

File: compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp [hunk 1]
```diff
@@ -63,6 +63,7 @@ static void buildMHLOInputConversionPassPipelineImpl(OpPassManager &passManager,
   passManager.addNestedPass<func::FuncOp>(mlir::createCanonicalizerPass());
   passManager.addNestedPass<func::FuncOp>(
       mhlo::createLegalizeControlFlowPass());
+  passManager.addNestedPass<func::FuncOp>(createEmitDefaultIREEABIPass());
```
Comment (benvanik) [inline compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp]: shouldn't you only need the one below?

**Review by benvanik — CHANGES_REQUESTED** [review]
Can you drop it from tosa for now? It's only required if you're going to use the python bindings and is more of an output/runtime option than one that should be determined by the input dialect. Making it default on all paths will make it really hard to make optional in the future as tests/etc will start relying on it - today having tosa not use it keeps us honest.

It'd still be nice if we didn't require it at all - would be good at some point to confirm that it's as load bearing as assumed on the python bindings path. Ideally this would be an option we expose for hosting runtimes/users to set when they need it and if we do default it on we have a flag to strip it that is tested to work.

**Comment by stellaraccident** [comment]
> It'd still be nice if we didn't require it at all - would be good at some point to confirm that it's as load bearing as assumed on the python bindings path. Ideally this would be an option we expose for hosting runtimes/users to set when they need it and if we do default it on we have a flag to strip it that is tested to work.

It's useful for tooling generally as it captures enough information to form a structurally valid call. Fuzzing, testing, etc. The need was independently discovered and I saw it/mentioned that we had something like that.

**Comment by jpienaar** [comment]
Updated to remove from TOSA path for now - good idea to have it be an option.

**Comment by ScottTodd** [comment]
What's the status of this PR? I see it in my review queue, but it is pretty old.

**Comment by benvanik** [comment]
Closing stale PR.

**Pull request closed without merging**